lang/parser/src/lexer/mod.rs

```diff
@@ -0,0 +1,136 @@
+use std::fmt;
+
+use logos::{Logos, SpannedIter};
+use num_bigint::BigUint;
+
+#[derive(Default, Debug, Clone, PartialEq)]
+pub enum LexicalError {
+    #[default]
+    InvalidToken,
+}
+
+impl fmt::Display for LexicalError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+#[derive(Logos, Clone, Debug, PartialEq)]
+#[logos(skip r"\s*", skip r"--(([^ \n\r]| [^\|\n\r])[^\n\r]*)?[\n\r]*", error = LexicalError)]
+pub enum Token {
+    // Keywords
+    //
+    //
+    #[token("data")]
+    Data,
+    #[token("codata")]
+    Codata,
+    #[token("def")]
+    Def,
+    #[token("codef")]
+    Codef,
+    #[token("let")]
+    Let,
+    #[token("match")]
+    Match,
+    #[token("as")]
+    As,
+    #[token("comatch")]
+    Comatch,
+    #[token("absurd")]
+    Absurd,
+    #[token("..absurd")]
+    DotsAbsurd,
+    #[token("Type")]
+    Type,
+
+    // Parens, Braces and Brackets
+    //
+    //
+    #[token("(")]
+    LParen,
+    #[token(")")]
+    RParen,
+    #[token("{")]
+    LBrace,
+    #[token("}")]
+    RBrace,
+    #[token("[")]
+    LBracket,
+    #[token("]")]
+    RBracket,
+
+    // Symbols
+    //
+    //
+    #[token(";")]
+    Semicolon,
+    #[token(":=")]
+    ColonEq,
+    #[token("=>")]
+    DoubleRightArrow,
+    #[token(",")]
+    Comma,
+    #[token(":")]
+    Colon,
+    #[token(".")]
+    Dot,
+    #[token("?")]
+    QuestionMark,
+    #[token("->")]
+    RightArrow,
+    #[token("\\")]
+    Backslash,
+    #[token("#")]
+    Hash,
+    #[token("_")]
+    Underscore,
+
+    // Names
+    //
+    //
+    #[regex(r"[A-ZΑ-Ω𝔹ℕ𝕍∃∀×][a-zα-ωA-ZΑ-Ω0-9_]*['⁺⁻₀₁₂₃₄₅₆₇₈₉₊₋]*", |lex| lex.slice().to_string())]
```

Comment:
Maybe we can use this as an opportunity to reconsider these :) Independent from this PR, though.

Reply:
Yes, we definitely have to reconsider those. This would be a good occasion to learn about all the different Unicode character classes :>